execution-engine/src/native_modules/darknet_inference.rs

```diff
@@ -0,0 +1,238 @@
+//! A native module for ML inference on Darknet.
+//! Takes an input image and various parameters, feeds it to the model and
+//! outputs a list of detected objects.
+//!
+//! ## Authors
+//!
+//! The Veracruz Development Team.
+//!
+//! ## Licensing and copyright notice
+//!
+//! See the `LICENSE_MIT.markdown` file in the Veracruz root directory for
+//! information on licensing and copyright.
+
+use crate::fs::{FileSystem, FileSystemResult};
+use crate::native_modules::common::Service;
+use darknet::{BBox, Detection, Image, Network};
+use lazy_static::*;
+use serde::Deserialize;
+use std::cmp::Ordering;
+use std::env::{current_dir, set_current_dir};
+use std::fmt::Write as _;
+use std::fs::{create_dir, create_dir_all, read_to_string, remove_dir_all, File};
+use std::io::Write;
+use std::path::{Path, PathBuf};
+use wasi_types::ErrNo;
+
+lazy_static! {
+    // Current directory before moving to the module's directory
+    static ref WORK_DIR: PathBuf = current_dir().unwrap_or(PathBuf::from(""));
+
+    // The native module executes in a dedicated directory to mitigate data
+    // leaks
+    static ref MODULE_DIR: PathBuf = WORK_DIR.clone().join("darknet_inference_service");
+}
+
+/// Module's API.
+#[derive(Deserialize, Debug)]
+pub(crate) struct DarknetInferenceService {
```

Comment:
This is a repeat of the stuff in `rust-examples`? I think we should have this in a shared file somewhere (but I don't know where that would be). If for some reason that's not possible or practicable there should be a comment explaining where the other copy is so that people know that they have to update both if it has to be changed.

Reply:
There is a lot of repeat, though it sounds like a bad idea to link the examples against Veracruz, they should be self-contained / easy to build.
Adding a comment to refer to the structure expected by the native module sounds like a good idea :+1: